packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -199,56 +221,179 @@ class AdaptiveScaffold extends StatefulWidget {
   /// Callback function for when the index of a [NavigationRail] changes.
   static WidgetBuilder emptyBuilder = (_) => const SizedBox();
 
+  static Builder toRailFromDestinations({
+    required List<NavigationDestination> destinations,
+    double width = 72,
+    int selectedIndex = 0,
+    bool extended = false,
+    Color backgroundColor = Colors.transparent,
+    Widget? leading,
+    Widget? trailing,
+    Function(int)? onDestinationSelected,
+    IconThemeData selectedIconTheme = const IconThemeData(color: Colors.black),
+    IconThemeData unselectedIconTheme =
+        const IconThemeData(color: Colors.black),
+    TextStyle selectedLabelTextStyle = const TextStyle(color: Colors.black),
+    NavigationRailLabelType labelType = NavigationRailLabelType.none,
+  }) {
+    return toNavigationRail(
+        width: width,
+        labelType: labelType,
+        leading: leading,
+        trailing: trailing,
+        onDestinationSelected: onDestinationSelected,
+        backgroundColor: backgroundColor,
+        extended: extended,
+        selectedIndex: selectedIndex,
+        selectedIconTheme: selectedIconTheme,
+        unselectedIconTheme: unselectedIconTheme,
+        selectedLabelTextStyle: selectedLabelTextStyle,
+        destinations: destinations
+            .map((NavigationDestination e) => _toRailDestination(e))
+            .toList());
+  }
+
   /// Public helper method to be used for creating a [NavigationRail] from a
   /// list of [NavigationDestination]s. Takes in a [selectedIndex] property for
   /// the current selected item in the [NavigationRail] and [extended] for
   /// whether the [NavigationRail] is extended or not.
   static Builder toNavigationRail({
-    required List<NavigationDestination> destinations,
+    required List<NavigationRailDestination> destinations,
     double width = 72,
     int selectedIndex = 0,
     bool extended = false,
     Color backgroundColor = Colors.transparent,
     Widget? leading,
     Widget? trailing,
     Function(int)? onDestinationSelected,
+    IconThemeData selectedIconTheme = const IconThemeData(color: Colors.black),
+    IconThemeData unselectedIconTheme =
+        const IconThemeData(color: Colors.black),
+    TextStyle selectedLabelTextStyle = const TextStyle(color: Colors.black),
```

Comment:
You don't need these. They will be set by the developer using a NavigationRailTheme.